include/sot/core/logger.hh

```diff
@@ -0,0 +1,227 @@
+ * modify it under the terms of the GNU Lesser General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version.
+ * sot-torque-control is distributed in the hope that it will be
+ * useful, but WITHOUT ANY WARRANTY; without even the implied warranty
+ * of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.  You should
+ * have received a copy of the GNU Lesser General Public License along
+ * with sot-torque-control.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef __sot_torque_control_logger_H__
+#define __sot_torque_control_logger_H__
+
+/* --------------------------------------------------------------------- */
+/* --- API ------------------------------------------------------------- */
+/* --------------------------------------------------------------------- */
+
+#if defined (WIN32)
+#  if defined (logger_EXPORTS)
+#    define LOGGER_EXPORT __declspec(dllexport)
+#  else
+#    define LOGGER_EXPORT __declspec(dllimport)
+#  endif
+#else
+#  define LOGGER_EXPORT
+#endif
+
+
+/* --------------------------------------------------------------------- */
+/* --- INCLUDE --------------------------------------------------------- */
+/* --------------------------------------------------------------------- */
+
+#include <sot/core/signal-helper.hh>
+#include <sot/core/matrix-geometry.hh>
+#include <map>
+#include <iomanip> // std::setprecision
+#include "boost/assign.hpp"
+
+
+namespace dynamicgraph {
+  namespace sot {
+
+//#define LOGGER_VERBOSITY_INFO_WARNING_ERROR
+#define LOGGER_VERBOSITY_ALL
+
+#define SEND_MSG(msg,type)         sendMsg(msg,type,__FILE__,__LINE__)
+
+#ifdef LOGGER_VERBOSITY_ERROR
+  #define SEND_DEBUG_STREAM_MSG(msg)
+  #define SEND_INFO_STREAM_MSG(msg)
+  #define SEND_WARNING_STREAM_MSG(msg)
+  #define SEND_ERROR_STREAM_MSG(msg)    SEND_MSG(msg,MSG_TYPE_ERROR_STREAM)
+#endif
+
+#ifdef LOGGER_VERBOSITY_WARNING_ERROR
+  #define SEND_DEBUG_STREAM_MSG(msg)
+  #define SEND_INFO_STREAM_MSG(msg)\
+  #define SEND_WARNING_STREAM_MSG(msg)  SEND_MSG(msg,MSG_TYPE_WARNING_STREAM)
+  #define SEND_ERROR_STREAM_MSG(msg)    SEND_MSG(msg,MSG_TYPE_ERROR_STREAM)
+#endif
+
+#ifdef LOGGER_VERBOSITY_INFO_WARNING_ERROR
+  #define SEND_DEBUG_STREAM_MSG(msg)
+  #define SEND_INFO_STREAM_MSG(msg)     SEND_MSG(msg,MSG_TYPE_INFO_STREAM)
+  #define SEND_WARNING_STREAM_MSG(msg)  SEND_MSG(msg,MSG_TYPE_WARNING_STREAM)
+  #define SEND_ERROR_STREAM_MSG(msg)    SEND_MSG(msg,MSG_TYPE_ERROR_STREAM)
+#endif
+
+#ifdef LOGGER_VERBOSITY_ALL
+  #define SEND_DEBUG_STREAM_MSG(msg) SEND_MSG(msg,MSG_TYPE_DEBUG_STREAM)
+  #define SEND_INFO_STREAM_MSG(msg)   SEND_MSG(msg,MSG_TYPE_INFO_STREAM)
+  #define SEND_WARNING_STREAM_MSG(msg)  SEND_MSG(msg,MSG_TYPE_WARNING_STREAM)
+  #define SEND_ERROR_STREAM_MSG(msg)    SEND_MSG(msg,MSG_TYPE_ERROR_STREAM)
+#endif
```

Comment:
dynamic-graph already provides tool for logging. I am not against a discussion to change it. Is this intended in this PR ?

I did not see any mention of it so I guess it isn't. I think it is a bit overkill. It doesn't seem to be used a lot.

Reply:
Moreover, I cannot see the definition of some declared functions like `getLogger`. Did you forget a file ?
I may have missed it though.

Reply:
> (were they already there in 2015?)

Some yes, others no.

I fine with adding them here if there are people who does use them. I would prefer if they were in dynamic-graph though.

Reply:
> Moreover, I cannot see the definition of some declared functions like `getLogger`. Did you forget a file ?
> I may have missed it though.

Yes, I need to add logger.cpp, thanks.